### PR TITLE
Offcanvas as overlay layer

### PR DESCRIPTION
**Pull request by TomS-**
My apologise for this pull request being added a few times. I've been developing for a long time now however I'm very new to GitHub.

---
#1453, Having the ability to have off-canvas as an overlay can lead to more modern designs (Google Material Design) and as mentioned it can make the animation less choppy.

It’s a very quick and easy change, however to make this work with the animation I’ve had to change it from “display: none;” to “visibility: hidden;”.

What do you think about that?

### CODE REVIEW

File: src/js/core/offcanvas.js [hunk 1]
```diff
@@ -27,7 +28,7 @@
             element.addClass("uk-active");
 
             $body.css({"width": window.innerWidth - scrollbarwidth, "height": window.innerHeight}).addClass("uk-offcanvas-page");
-            $body.css((rtl ? "margin-right" : "margin-left"), (rtl ? -1 : 1) * (bar.outerWidth() * dir)).width(); // .width() - force redraw
+            if (!over) { $body.css((rtl ? "margin-right" : "margin-left"), (rtl ? -1 : 1) * (bar.outerWidth() * dir)).width(); // .width() - force redraw }
```
Comment (krokerke) [inline src/js/core/offcanvas.js]: the closing bracket '}' must be before the comment

**Comment by TomS-** [comment]
@acaron, I believe resize is a problem with uikit, I can fix this but it's a separate issue. I believe that code prevents scrolling on the main page when scrolling the sidebar so needs to be there at all times. I'll create a pull request for fixing the resize issue.

**Comment by vinoth3v** [comment]
any update on Offcanvas as overlay layer?

**Comment by jsieler** [comment]
Hm, like this one?
https://uikitplay.com/plays/FXsr

**Comment by vinoth3v** [comment]
Yes, yes. Possible now?

**Comment by jsieler** [comment]
@vinoth3v see code inside. It's just small CSS addition which makes it possible.

**Comment by vinoth3v** [comment]
@jsieler, It seems working! Thank you very much!

**Comment by jsieler** [comment]
@vinoth3v you are welcome!

**Comment by philoest** [comment]
UIKit version 2 has been moved to another branch: https://github.com/uikit/uikit/tree/v2/develop

Please update the target branch of your pull request.

**Pull request closed without merging**